packages/table/src/cell/editableCell.tsx

```diff
@@ -40,45 +41,81 @@ export interface IEditableCellProps extends ICellProps {
     onConfirm?: (value: string) => void;
 }
 
-export class EditableCell extends React.Component<IEditableCellProps, {}> {
-    private cellElement: HTMLElement;
+export interface IEditableCellState {
+    /**
+     * Stores the editing state of the cell
+     */
+    isEditing: boolean;
+}
+
+@PureRender
+export class EditableCell extends React.Component<IEditableCellProps, IEditableCellState> {
+    public constructor(props: IEditableCellProps, context?: any) {
+        super(props, context);
+        this.state = {
+            isEditing: false,
+        };
```

Comment:
the compiler writes a constructor for you and puts all those instance properties inside it. same thing with bound `private func = () => {}` properties.

from `/docs/docs/docs.js` (can't link to lines):
![image](https://cloud.githubusercontent.com/assets/464822/21906471/4dde7604-d8c0-11e6-8a29-2433dceec9cb.png)